src/strategies/uniswapv3/strategy-univ3-rebalance.sol

```diff
@@ -464,50 +429,86 @@ abstract contract StrategyRebalanceUniV3 {
+                    _token0ToTrade
+                );
+                // token0 is native
+            } else {
+                _nativeToTreasury += _token0ToTrade;
+            }
+        }
+
+        if (_amount1 > 0) {
+            uint256 _token1ToTrade = _amount1.mul(performanceTreasuryFee).div(performanceTreasuryMax);
+
+            if (tokenToNativeRoutes[address(token1)].length > 0) {
+                _nativeToTreasury += _swapUniV3WithPath(
+                    address(token1),
+                    tokenToNativeRoutes[address(token1)],
+                    _token1ToTrade
+                );
+                // token1 is native
+            } else {
+                _nativeToTreasury += _token1ToTrade;
+            }
+        }
+        if (_nativeToTreasury > 0) IERC20(native).safeTransfer(IControllerV2(controller).treasury(), _nativeToTreasury);
+    }
+
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes memory
+    ) public pure returns (bytes4) {
+        return this.onERC721Received.selector;
+    }
+
     function _balanceProportion(int24 _tickLower, int24 _tickUpper) internal {
         PoolVariables.Info memory _cache;
 
         _cache.amount0Desired = token0.balanceOf(address(this));
         _cache.amount1Desired = token1.balanceOf(address(this));
 
         //Get Max Liquidity for Amounts we own.
-        _cache.liquidity = pool.liquidityForAmounts(
-            _cache.amount0Desired,
-            _cache.amount1Desired,
-            _tickLower,
-            _tickUpper
+        (uint160 sqrtPriceX96, , , , , , ) = pool.slot0();
+        uint160 sqrtRatioAX96 = TickMath.getSqrtRatioAtTick(_tickLower);
+        uint160 sqrtRatioBX96 = TickMath.getSqrtRatioAtTick(_tickUpper);
+
+        _cache.liquidity = uint128(
+            LiquidityAmounts.getLiquidityForAmount0(sqrtRatioAX96, sqrtRatioBX96, _cache.amount0Desired).add(
+                LiquidityAmounts.getLiquidityForAmount1(sqrtRatioAX96, sqrtRatioBX96, _cache.amount1Desired)
+            )
         );
 
         //Get correct amounts of each token for the liquidity we have.
-        (_cache.amount0, _cache.amount1) = pool.amountsForLiquidity(
-            _cache.liquidity,
-            _tickLower,
-            _tickUpper
+        (_cache.amount0, _cache.amount1) = LiquidityAmounts.getAmountsForLiquidity(
+            sqrtPriceX96,
+            sqrtRatioAX96,
+            sqrtRatioBX96,
+            _cache.liquidity
         );
 
         //Determine Trade Direction
-        bool _zeroForOne;
-        if (_cache.amount1Desired == 0) {
-            _zeroForOne = true;
-        } else {
-            _zeroForOne = PoolVariables.amountsDirection(
-                _cache.amount0Desired,
-                _cache.amount1Desired,
-                _cache.amount0,
-                _cache.amount1
-            );
-        }
+        bool _zeroForOne = _cache.amount0Desired > _cache.amount0 ? true : false;
```

Comment:
I know you inherited this naming convention, but it's so confusing that `amount0Desired` is what you already have, and `amount0` is what you desire to obtain 😅